src/Driver.cpp

```diff
@@ -1,598 +1,558 @@
+{
+    base::samples::RigidBodyState new_pose;
+    new_pose.position[0] = usbl_pose.x;
+    new_pose.position[1] = usbl_pose.y;
+    new_pose.position[2] = usbl_pose.z;
 
-	    std::string buffer_as_string = std::string(rim.buffer.begin(), rim.buffer.end());
-	        std::cout << "trying parse remote\n";
+    return new_pose;
+}
 
-            current_position = UsblParser::parseRemotePosition(buffer_as_string);
-        } catch (ParseError e){
-	    	        std::cout << "there was a parse error, handling IM as normal IM\n";
+void Driver::sendIMPose(base::samples::RigidBodyState const &send_pose)
+{
 
-            //If its can't be parsed as remote Position it's a User IM.
-            receivedInstantMessages.push_back(rim);
-        }
-    } else {
-        receivedInstantMessages.push_back(rim);
-    }
-    //TODO hier kann die uhrzeit nicht abgefragt werden alternative?
-    //rim.time = getSystemTime();
-}
-size_t Driver::readInternal(uint8_t *buffer, size_t buffer_size){
-    size_t packet_size = readPacket(buffer, buffer_size, 3000, 3000);
-    std::string buffer_as_string = std::string(reinterpret_cast<char const*>(buffer));
-    if (packet_size){
-        if (UsblParser::isPacket(buffer_as_string) > 0){
-            if (!handleAsynchronousCommand(buffer_as_string)){
-                //The Packet is not handled as asynchronous Command,
-                //so there is a unhandled packet with size packet_size
-                return packet_size;
-            } else {
-                //The packet is handled already as asynchronous Command
-                //so there is no unhandled packet
-                return 0;
-            }
+}
 
-        } else {
-            std::cout << " Buffer is burst data. Ignoring burst data when reading internal." << buffer << std::endl;
-            return 0;
-        }
-    }
-    return 0;
-}
-bool Driver::handleAsynchronousCommand(std::string buffer_as_string){
-    std::cout << "usbldriver handleAsync\n";
-    switch (UsblParser::parseAsynchronousCommand(buffer_as_string)){
-        case NO_ASYNCHRONOUS:
-            return false;
-        case DELIVERY_REPORT:
-            std::cout << "There was a delivery report" << std::endl;
-            incomingDeliveryReport(buffer_as_string);
-            return true;
-        case INSTANT_MESSAGE:
-            std::cout << "There was a instant message" << std::endl;
-            incomingInstantMessage(buffer_as_string);
-            return true;
-        case CANCELEDIM:
-            std::cout << "There is a CANCLEDIM" << std::endl;
-            cancelIm(buffer_as_string);
-            return true;
-        case USBLLONG:
-            incomingPosition(buffer_as_string);
-        case USBLANGLE:
-            std::cout << "There is a USBL Asynch Message" << std::endl;
-            //TODO Handle this, when possible (unknown protocol)
-            return true;
-    }
-    return false;
+void Driver::goSurface(void)
```

Comment:
This is the evologics driver, stick to what the device does natively.

After that, you could create a auv_uwmodem_commands library or something that would sit on top to provide a generic command/status library